File: go/fiber/internal/modules/client/module.go
package client

import (
	"database/sql"
	"tcc/internal/common/enum"
	"tcc/internal/common/middleware"
	auth "tcc/internal/modules/auth/middleware"

	"github.com/gofiber/fiber/v2"
)

func ClientModule(router *fiber.App, db *sql.DB) {
	repository := ClientRepository{
		db: db,
	}
	service := ClientService{
		repository: repository,
	}
	controller := ClientController{
		service: service,
	}

	group := router.Group("/clientes")
	{
		group.Get("/cpf/:cpf", auth.AuthMiddleware(), middleware.RolesPermittedMiddleware(enum.MANAGER), controller.FindByCPF)
		group.Post("", controller.Create)
	}
}

File: python/fastapi/src/app/modules/clients/module.py
from app.common.classes.base_module import BaseModule

# Controladores
from .controller import ClientsController

# Serviços
from .service import ClientsService

# Repositório
from .repository import ClientsRepository

# Middlewares
from ..auth.middlewares.auth import auth_middleware
from app.common.middlewares.roles_permitted import roles_permitted_middleware

# Types
from app.common.enums import RoleEnum

class ClientsModule(BaseModule):
  def __init__(self):
    _repository = ClientsRepository()
    _service = ClientsService(_repository)
    self._controller = ClientsController(_service)
    super().__init__(prefix="clientes")

  def _routes(self):
    self._router.post(self._controller.create)
    self._router.get(self._controller.findByCPF, path="/cpf/{cpf}", dependencies=[auth_middleware, roles_permitted_middleware(RoleEnum.MANAGER)])